crates/rattler_installs_packages/src/wheel.rs

```diff
@@ -0,0 +1,877 @@
+        let mut candidates = top_level_names.into_iter().filter(|dir_name| {
+            let Some(candidate) = dir_name.strip_suffix(suffix) else {
+                return false;
+            };
+            let Some((candidate_name, candidate_version)) = candidate.rsplit_once('-') else {
+                return false;
+            };
+
+            let Ok(candidate_name) = PackageName::from_str(candidate_name) else {
+                return false;
+            };
+            let Ok(candidate_version) = Version::from_str(candidate_version) else {
+                return false;
+            };
+
+            &candidate_name == name && &candidate_version == version
+        });
+
+        // Get the first candidate
+        let candidate = match candidates.next() {
+            Some(candidate) => candidate,
+            None => return Ok(None),
+        };
+
+        // Error out if there are multiple directories
+        if candidates.next().is_some() {
+            return Err(WheelVitalsError::MultipleSpecialDirs(suffix.to_owned()));
+        }
+
+        Ok(Some(candidate))
+    }
+
+    async fn get_lazy_vitals(
+        name: &WheelName,
+        stream: &mut AsyncHttpRangeReader,
+    ) -> Result<(Vec<u8>, WheelCoreMetadata), WheelVitalsError> {
+        // Make sure we have the back part of the stream.
+        stream
+            .prefetch(stream.len().saturating_sub(16384)..stream.len())
+            .await;
+
+        // Construct a zip reader to uses the stream.
+        let mut reader = ZipFileReader::new(stream.compat())
+            .await
+            .map_err(|err| WheelVitalsError::from_async_zip("/".into(), err))?;
+
+        // Collect all top-level filenames
+        let top_level_names = reader
+            .file()
+            .entries()
+            .iter()
+            .filter_map(|e| e.entry().filename().as_str().ok())
+            .map(|filename| {
+                filename
+                    .split_once(['/', '\\'])
+                    .map_or_else(|| filename, |(base, _)| base)
+            })
+            .collect::<HashSet<_>>();
+
+        // Determine the name of the dist-info directory
+        let dist_info = Wheel::find_special_wheel_dir(
+            top_level_names.iter().copied(),
+            &name.distribution,
+            &name.version,
+            ".dist-info",
+        )?
+        .ok_or(WheelVitalsError::DistInfoMissing)?
+        .to_owned();
+
+        let metadata_path = format!("{dist_info}/METADATA");
+        let (metadata_idx, metadata_entry) = reader
+            .file()
+            .entries()
+            .iter()
+            .enumerate()
+            .find(|(_, p)| p.entry().filename().as_str().ok() == Some(metadata_path.as_str()))
+            .ok_or(WheelVitalsError::MetadataMissing)?;
+
+        // Get the size of the entry plus the header + size of the filename. We should also actually
+        // include bytes for the extra fields but we don't have that information.
```

Comment:
What happens now that we don't know this?

Reply:
We cross our fingers that the entry doesnt include extra fields. But because we also align to the bufreader size most likely we dont need to do another range request anyway. In the worst case another range request is performed.